Modules/Net/readers.swift

```diff
@@ -93,22 +98,100 @@ internal class UsageReader: Reader<Network_Usage> {
         }
         freeifaddrs(interfaceAddresses)
         
-        if self.usage.upload != 0 && self.usage.download != 0 {
-            self.usage.upload = upload - self.usage.upload
-            self.usage.download = download - self.usage.download
+        if !verifiedUsageDataApproach {
+            if didNativeApproachReportNonZeroDownload || nonZeroUpdatesCount >= 3 {
+                if !didNativeApproachReportNonZeroDownload {
+                    shouldUseProcessesUsageData = true
+                }
+                verifiedUsageDataApproach = true
+            } else {
+                let processesUsage = allProcessesTotalUsage()
+                let accumulatedDownload = processesUsage?.download ?? 0
+                if download > 0 || accumulatedDownload > 0 {
+                    nonZeroUpdatesCount += 1
+                }
+                if download > 0 {
+                    didNativeApproachReportNonZeroDownload = true
+                }
+            }
         }
         
-        if self.usage.upload < 0 {
-            self.usage.upload = 0
-        }
-        if self.usage.download < 0 {
-            self.usage.download = 0
+        let accumulatedDownload: Int64
+        if shouldUseProcessesUsageData {
+            let processesUsage = allProcessesTotalUsage()
+            accumulatedDownload = processesUsage?.download ?? 0
+        } else {
+            accumulatedDownload = download
         }
         
+        self.usage.upload = max(upload - self.usage.upload, 0)
+        self.usage.download = max(accumulatedDownload - self.usage.download, 0)
+        
         self.callback(self.usage)
         
         self.usage.upload = upload
-        self.usage.download = download
+        self.usage.download = accumulatedDownload
+    }
+    
+    private func allProcessesTotalUsage() -> Network_Usage? {
```

Comment:
Also, there is no need to have a `Network_Usage` struct here. I propose to return a tuple.